app_dart/lib/src/service/cache_service.dart

```diff
@@ -95,10 +129,46 @@ class CacheService {
     return entry.set(value, ttl);
   }
 
+  /// Set [value] for [key] in the subcache [subcacheName] with [ttl] but
+  /// enforce locking accessing.
+  ///
+  /// Note: these methods are intended to prevent issues around race conditions
+  /// when storing and retrieving github tokens. Care should be taken to use the
+  /// locking methods together when accessing data from an entity using the
+  /// cache.
+  Future<Uint8List?> setWithLocking(
+    String subcacheName,
+    String key,
+    Uint8List? value, {
+    Duration ttl = const Duration(minutes: 1),
+  }) async {
+    await m.acquireWrite();
+    try {
+      return set(
+        subcacheName,
+        key,
+        value,
+        ttl: ttl,
+      );
+    } finally {
+      m.release();
+    }
+  }
+
   /// Clear the value stored in subcache [subcacheName] for key [key].
-  Future<void> purge(String subcacheName, String key) {
-    final Cache<Uint8List> subcache = cache.withPrefix(subcacheName);
-    return subcache[key].purge(retries: maxCacheGetAttempts);
+  ///
+  /// Note: these methods are intended to prevent issues around race conditions
+  /// when storing and retrieving github tokens. Care should be taken to use the
+  /// locking methods together when accessing data from an entity using the
+  /// cache.
+  Future<void> purge(String subcacheName, String key) async {
+    await m.acquireWrite();
```

Comment:
Why does purge need a write lock?

Reply:
Purge is a write so it is good in the case that something is writing through locking that we prevent that action while attempting to purge the cache. That way we don't immediately write then wipe the value.

Reply:
Up to you if you don't think we need it.